### PR TITLE
Adding ability to import/reference external tf states

### PR DESCRIPTION
**Pull request by juandiegopalomino**
# Description
See the docs

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
Manually

### CODE REVIEW

File: modules/external_state/external-state.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "$id": "https://app.runx.dev/modules/external-state",
+  "type": "object",
+  "description": "Adds the ability to refer to external, preexisting terraform states in opta.",
+  "properties": {
+    "backend_type": {
+      "type": "string",
+      "description": "The type of backend this external state is using"
+    },
+    "config": {
+      "type": "string",
```
Comment (RemyDeWolf) [inline modules/external_state/external-state.json:11]: we can use the object type like what we do with datadog or helm charts values
https://github.com/run-x/opta/blob/main/modules/datadog/datadog.json#L15-L18

File: modules/external_state/external-state.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "$id": "https://app.runx.dev/modules/external-state",
+  "type": "object",
+  "description": "Adds the ability to refer to external, preexisting terraform states in opta.",
+  "properties": {
+    "backend_type": {
+      "type": "string",
+      "description": "The type of backend this external state is using"
```
Comment (RemyDeWolf) [inline modules/external_state/external-state.json:8]: can we also add the link here https://www.terraform.io/language/settings/backends/configuration#backend-types

File: modules/external_state/external-state.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+weight: 1
+description: Adds the ability to refer to external, preexisting terraform states in opta.
+---
+
+This module grants users the ability to import outputs from 
+[external terraform backends](https://www.terraform.io/language/settings/backends) (e.g. pre-existing
+or created outside of opta, or another Opta service/environment). All the terraform outputs of that state
+will be captured and referenceable under the `outputs` output for this module. 
+
+For instance, you can refer values such as the name of an S3 bucket created in another Opta service like so:
```
Comment (RemyDeWolf) [inline modules/external_state/external-state.md:15]: I can see some possible confusion for some users between the S3 used for the backend and the S3 bucket used in this example. But they are different things.

Can we use another example? may be name of EKS cluster, Database name or VPC id these are usually common.

File: modules/external_state/external-state.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+The service will now have an environment variables known as SOURCE_BUCKET which could be the name of the
+bucket they wish grab data from (assuming IAM permissions are configured externally for simplicity).
+
+{{% alert title="NOTE" color="orange" %}}
+Usage is reliant on what the external state outputs and names. Please make sure that the data you seek
+is both outputted and has the expected name. If the state is created by Opta you can find the outputs
+and name by running opta apply on that state's manifest yaml.
+{{% /alert %}}
```
Comment (RemyDeWolf) [inline modules/external_state/external-state.md:47]: what will happen if the output name is not found?

File: modules/external_state/external-state.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+---
+title: "external-state"
+linkTitle: "external-state"
+date: 2021-07-21
+draft: false
+weight: 1
+description: Adds the ability to refer to external, preexisting terraform states in opta.
```
Comment (RemyDeWolf) [inline modules/external_state/external-state.md:7]: random question, can this module be used when referring to a stack managed by opta?

I don't know if it's an interesting use case.

File: modules/external_state/tf_module/main.tf [hunk 6]
```diff
@@ -0,0 +1,5 @@
+data "terraform_remote_state" "remote" {
+  backend = var.backend_type
+
+  config = var.config
+}
```
Comment (RemyDeWolf) [inline modules/external_state/tf_module/main.tf:5]: 👍 woo, that was it!

File: modules/external_state/external-state.md (hunk 5 above)
Comment (juandiegopalomino) [reply modules/external_state/external-state.md:7]: `(e.g. pre-existing
or created outside of opta, or another Opta service/environment)`

**Review by RemyDeWolf — APPROVED** [review]

**Pull request merged**